xrpl4j-integration-tests/src/test/java/org/xrpl/xrpl4j/tests/DidIT.java

```diff
@@ -0,0 +1,270 @@
+      did.ownerKeyPair().publicKey().deriveAddress());
+
+    XrpCurrencyAmount fee = FeeUtils.computeNetworkFees(xrplClient.fee()).recommendedFee();
+    DidDelete didDelete = DidDelete.builder()
+      .account(did.ownerKeyPair().publicKey().deriveAddress())
+      .sequence(sourceAccountInfo.accountData().sequence())
+      .fee(fee)
+      .lastLedgerSequence(sourceAccountInfo.ledgerIndexSafe().plus(UnsignedInteger.valueOf(4)).unsignedIntegerValue())
+      .signingPublicKey(did.ownerKeyPair().publicKey())
+      .build();
+
+    TransactionResult<DidDelete> result = this.signSubmitAndWait(didDelete, did.ownerKeyPair(), DidDelete.class);
+
+    logger.info(ObjectMapperFactory.create().writerWithDefaultPrettyPrinter().writeValueAsString(result));
+    assertThat(result.metadata()).isNotEmpty();
+    List<DeletedNode<MetaDidObject>> deletedNodes = result.metadata().get().affectedNodes().stream()
+      .filter(node -> node.ledgerEntryType().equals(MetaLedgerEntryType.DID))
+      .filter(node -> DeletedNode.class.isAssignableFrom(node.getClass()))
+      .filter(node -> MetaDidObject.class.isAssignableFrom(((DeletedNode<?>) node).finalFields().getClass()))
+      .map(node -> (DeletedNode<MetaDidObject>) node)
+      .collect(Collectors.toList());
+
+    assertThat(deletedNodes.size()).isEqualTo(1);
+    DeletedNode<MetaDidObject> deletedNode = deletedNodes.get(0);
+    MetaDidObject finalFields = deletedNode.finalFields();
+    assertThat(finalFields.didDocument()).isNotEmpty().isEqualTo(did.object().didDocument());
+    assertThat(finalFields.uri()).isNotEmpty().isEqualTo(did.object().uri());
+    assertThat(finalFields.data()).isNotEmpty().isEqualTo(did.object().data());
+    assertThat(finalFields.account()).isNotEmpty().get().isEqualTo(sourceAccountInfo.accountData().account());
+    assertThat(finalFields.flags()).isEqualTo(Flags.UNSET);
+
+    List<DidObject> accountObjects = this.getValidatedAccountObjects(
+      sourceAccountInfo.accountData().account(),
+      DidObject.class
+    );
+
+    assertThat(accountObjects).asList().isEmpty();
+  }
+
+  private TestDid createNewDid() throws JsonRpcClientErrorException, JsonProcessingException {
+    KeyPair sourceKeyPair = this.createRandomAccountEd25519();
+
+    AccountInfoResult sourceAccountInfo = this.scanForResult(
+      () -> this.getValidatedAccountInfo(sourceKeyPair.publicKey().deriveAddress())
+    );
+
+    XrpCurrencyAmount fee = FeeUtils.computeNetworkFees(xrplClient.fee()).recommendedFee();
+
+    DidSet didSet = DidSet.builder()
+      .account(sourceAccountInfo.accountData().account())
+      .sequence(sourceAccountInfo.accountData().sequence())
+      .fee(fee)
+      .lastLedgerSequence(sourceAccountInfo.ledgerIndexSafe().unsignedIntegerValue().plus(UnsignedInteger.valueOf(4)))
+      .signingPublicKey(sourceKeyPair.publicKey())
+      .data(DidData.of("617474657374"))
+      .uri(DidUri.of("6469645F6578616D706C65"))
```

Comment:
What does rippled do if this isn't HEX?

Reply:
See comment above -- I can't actually submit a transaction with a non-hex value for any of these fields because the binary codec will throw an exception.

You could talk me into validating that `DidData`, `DidUri`, and `DidDocument` are hex encoded in those objects so developers get better exceptions...

Reply:
Maybe make a comment in https://github.com/XRPLF/xrpl4j/issues/500 to improve the binary codec errors.